graphium/fingerprinting/data.py

```diff
@@ -0,0 +1,216 @@
+from typing import Any, List, Dict, Literal, Union
+
+import os
+
+import torch
+
+import pandas as pd
+import numpy as np
+
+from pytorch_lightning import LightningDataModule
+
+from torch.utils.data import Dataset, DataLoader
+
+from graphium.data.datamodule import MultitaskFromSmilesDataModule, ADMETBenchmarkDataModule
+from graphium.trainer.predictor import PredictorModule
+from graphium.fingerprinting.fingerprinter import Fingerprinter
+
+
+class FingerprintDataset(Dataset):
+    def __init__(
+            self,
+            smiles: List[str],
+            labels: torch.Tensor,
+            fingerprints: Dict[str, torch.Tensor],
+    ):
+        self.smiles = smiles
+        self.labels = labels
+        self.fingerprints = fingerprints
+
+    def __len__(self):
+        return len(self.smiles)
+
+    def __getitem__(self, index):
+        fp_list = []
+        for val in self.fingerprints.values():
+            fp_list.append(val[index])
+        return fp_list, self.labels[index]
+
+
+class FingerprintDatamodule(LightningDataModule):
```

Comment:
Missing docs for parameters of that class. You can just copy/paste from the MultiTaskDatamodule

Reply:
Also fixed [here](https://github.com/datamol-io/graphium/pull/529/commits/d9aa407d774934e170e8d8af7028155b676b5981#diff-a660520d63c09919ce7b7d4450b3edf709bba2c5d21fa421a52fa4d1fff7a9e3).